nonblock-rpc/rust/core/src/router.rs

```diff
@@ -0,0 +1,95 @@
+use std::collections::HashMap;
+
+use serde::Serialize;
+
+pub struct Router {
+    pub routes: HashMap<String, Box<dyn Handler + 'static + Send + Sync>>,
+}
+
+impl Router {
+    pub fn new() -> Self {
+        Router {
+            routes: HashMap::new(),
+        }
+    }
+
+    pub fn add(mut self, name: &str, handler: impl Handler + 'static + Send + Sync) -> Self {
+        self.routes.insert(name.to_string(), Box::new(handler));
+        self
+    }
+
+    pub fn call(&self, name: &str, args: Vec<String>) -> String {
+        match self.routes.get(name) {
+            Some(handler) => handler.call(args),
+            None => "Unknown function".to_string(),
+        }
+    }
```

Comment:
`Unknown function` should be wrapped in an error type. The `call()` method should return `Result<String>` instead.